common/src/main/java/org/jboss/pnc/common/Configuration.java

```diff
@@ -33,19 +37,25 @@
      * @return Loaded configuration
      * @throws ConfigurationParseException Thrown if configuration file couldn't be loaded or parsed
      */
-    public T getModuleConfig(Class<T> moduleClass) throws ConfigurationParseException {
+    @SuppressWarnings("unchecked")
+    public <T extends AbstractModuleConfig> T getModuleConfig(Class<T> moduleClass) throws ConfigurationParseException {
```

Comment:
I believe we may end up with many configurations here, since there is no synchronization...

I believe we have a couple of options:
* Double check locking pattern (which is also broken :D)
* Use Guava's Cache Loader (or Infinispan!)
* Use EJB @Singleton with locking

The fastest is the third one, but I think the best one is the second.